odoo/addons/base/models/ir_actions_report.py

```diff
@@ -658,6 +669,8 @@ def barcode(self, barcode_type, value, **kwargs):
         }
         kwargs = {k: validator(kwargs.get(k, v)) for k, (v, validator) in defaults.items()}
         kwargs['humanReadable'] = kwargs.pop('humanreadable')
+        if _DEFAULT_BARCODE_FONT:
+            kwargs['fontName'] = _DEFAULT_BARCODE_FONT
```

Comment:
It seems that it prevent to choose a custom font 🤔 
_DEFAULT_BARCODE_FONT is falsy only when you cannot set the ascent and descent

Reply:
validating the kwargs with defaults prevent use of non-default keys, so no way to choose another font
moreover, these defaults are a superset of the ones in BarcodeConverter.value_to_html